src/ngraph/frontend/onnx_import/op/dequantize_linear.cpp

```diff
@@ -0,0 +1,94 @@
+
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <memory>
+
+#include "exceptions.hpp"
+#include "ngraph/axis_set.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/dequantize.hpp"
+#include "ngraph/shape.hpp"
+#include "quantize_linear.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace set_1
+            {
+                NodeVector dequantize_linear(const Node& node)
+                {
+                    NodeVector inputs{node.get_ng_inputs()};
+                    std::shared_ptr<ngraph::Node> x = inputs.at(0);
+                    std::shared_ptr<ngraph::Node> y_scale = inputs.at(1);
+                    std::shared_ptr<ngraph::Node> y_zero_point = inputs.at(2);
+
+                    bool has_axis = false;
+                    Shape x_shape = x->get_shape();
+                    Shape y_scale_shape = y_scale->get_shape();
+                    Shape y_zero_point_shape = y_zero_point->get_shape();
+                    AxisSet axis_set{};
+
+                    try
+                    {
+                        std::int64_t axis = node.get_attribute_value<std::int64_t>("axis");
+                        axis_set = AxisSet{static_cast<std::size_t>(axis)};
+                        has_axis = true;
+                    }
+                    catch (const error::node::UnknownAttribute&)
+                    {
+                    }
+
+                    // Per `axis` quantization. Input ‘scale’ and ‘zero_point’ must be 1-D tensors
+                    if (has_axis)
+                    {
+                        std::size_t axis_dim_value = x_shape.at(*std::begin(axis_set));
+
+                        ASSERT_VALID_ARGUMENT(
+                            node,
+                            (y_scale_shape.size() == 1 && y_scale_shape.at(0) == axis_dim_value))
+                            << "y_scale must be 1D tensor with size equal to: " << axis_dim_value;
+                        ASSERT_VALID_ARGUMENT(node,
+                                              (y_zero_point_shape.size() == 1 &&
+                                               y_zero_point_shape.at(0) == axis_dim_value))
+                            << "y_zero_point must be 1D tensor with size equal to: "
+                            << axis_dim_value;
+                    }
+                    // Per tensor quantization. Input ‘scale’ and ‘zero_point’ must be scalars
+                    else
+                    {
+                        ASSERT_VALID_ARGUMENT(node, y_scale_shape.size() == 0)
+                            << "y_scale must be a scalar if no axis is provided.";
+                        ASSERT_VALID_ARGUMENT(node, y_zero_point_shape.size() == 0)
+                            << "y_zero_point must be a scalar if no axis is provided.";
+                    }
+
+                    return {std::make_shared<ngraph::op::Dequantize>(
```

Comment:
dequantize in CPU/IAT backend expects constant zero, we need below check,
```
if (x->get_element_type() != y_zero_point->get_element_type())                        
{                                                                                     
    // Currently only support Constant node.                                          
    ASSERT_IS_SUPPORTED(node, y_zero_point->is_constant())                            
        << "doesn't support zero point input of other type than Constant.";           
                                                                                      
    auto y_zp_constant = std::static_pointer_cast<ngraph::op::Constant>(y_zero_point);
                                                                                      
    y_zero_point = ngraph::op::Constant::create<std::uint8_t>(                        
        x->get_element_type(),                                                        
        y_zero_point->get_shape(),                                                    
        y_zp_constant->get_vector<std::uint8_t>());
}
```

Reply:
after discussion with @adstraw , it seems this check can be done on CPU/IAT backend side.

src/ngraph/runtime/cpu/pass/cpu_assignment.cpp:754

Reply:
dequantize core API expects `
x->get_element_type() == y_zero_point->get_element_type()` , If this is not the case, we will need to recreate zero with same type as in above snippet. @adstraw maybe we need above check in this PR ?

Reply:
Is converting `offset` to `data` element type in this CPU pass is a good idea?
Or should we convert it in the onnx bridge?
If `offset == 0` is checked in pass already.

Reply:
`convert it in the onnx bridge` is better, as this is API constraint/requirement.